watchdog/handler.go

```diff
@@ -299,8 +299,11 @@ func makeRequestHandler(config *WatchdogConfig) func(http.ResponseWriter, *http.
 		case
 			http.MethodPost,
 			http.MethodPut,
+			http.MethodPatch,
 			http.MethodDelete,
-			http.MethodGet:
+			http.MethodGet,
+			http.MethodOptions,
```

Comment:
Options and head were left off the original PR on purpose since no existing watchdog-based functions would know how to respond to that and even if they did they cannot return the proper HTTP status codes for not supported/implemented with the classic watchdog.

I'd put those two out of scope for now or comment out for this PR.